solutions/hw5/treap.py

```diff
@@ -0,0 +1,122 @@
+from collections import MutableMapping
+from typing import Iterator, Callable, Any
+from random import randint
+
+
+class Node:
+    priority: int
+
+    def __init__(self, key, value):
+        self.key = key
+        self.value = value
+        self.left_child = None
+        self.right_child = None
+        self.priority = randint(0, 100)
+
+    def inorder(self) -> list:
```

Comment:
I think it is better to use generators for the orders. Also, it would be great if you add two types of orders instead one